### PR TITLE
JS interop updates

### LINKED ISSUE #26980: {ASSEMBLY NAME} appears to be optional when working within the same assembly
The documentation doesn't make clear that {ASSEMBLY NAME} does not need to be specified when working within the same assembly. Perhaps this is common knowledge but it was not immediately clear to me.

dotNetHelper.invokeMethodAsync("MethodName", data); is how we are using it in our codebase.

I believe the documentation would be clearer if some mention were made that {ASSEMBLY NAME} is optional in certain cases.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 2964d558-cd6d-7421-f7b9-cb939792c2cf
* Version Independent ID: a3517e56-69fc-1ec7-b55a-dbb54d68a807
* Content: [Call .NET methods from JavaScript functions in ASP.NET Core Blazor](https://docs.microsoft.com/en-us/aspnet/core/blazor/javascript-interoperability/call-dotnet-from-javascript?view=aspnetcore-6.0)
* Content Source: [aspnetcore/blazor/javascript-interoperability/call-dotnet-from-javascript.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/blazor/javascript-interoperability/call-dotnet-from-javascript.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-blazor**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #26980

Thanks @contractcooker! 🚀 ...

* I finally reached your issue. Sorry for the delay. It was a busy .NET 7 release season late last year, and I'm just now wrapping up last year's backlog issues.
* Two things that I found out improving the JS interop coverage:
  * I found a bug in one of the examples that didn't prevent it from running ... the example just wasn't composed correctly. I found one line in the article's disposal example for instance methods that required an update. (I wish that I had known about this because I would've marked your issue as a 🪲 and worked it last year.)
  * Also, I found a blasted ***spelling error*** 😈 in one of the code examples 🙈 ... that HEX array byte array support example that writes out a Serenity movie quote via JS interop ... that code example misspelled one of the words in the quote. ***Arrrrrgh!*** 😠 Anyway, I spotted it, updated the HEX, and it will be fine going forward.
* Assembly names aren't optional in the static method call case. The call throws at runtime if the assembly name isn't the first parameter. In the instance cases, the assembly name isn't a parameter. I've improved the section on instance method calls. I'm curious about the circumstances where you found that the assembly name is optional and the call is invoked either way. You didn't state exactly which JS interop scenario that you were working with at the time. If you found one or more cases where it looks optional and instance method calls execute either way, please put up a GH repro project or paste your code examples without and with the assembly name parameter into an issue comment back on the original issue that you opened because I'd like to research it further and see what I'm failing to understand ... I'd reach out to Javier to ask for a clarification on it for another update to this article. Currently tho, all of the instance method examples are going to show ***no*** assembly name parameter. I've made a note to follow-up on this PR with a series of example tests to make sure that the sample apps that provide this code are all working well.

### CODE REVIEW

**Comment by guardrex** [comment]
Actually, let's hold here for a sec and ask @MackinnonBuck to comment on this ...

Mackinnon, can you clarify what I think I'm seeing for calling static and instance .NET methods with JS interop? ...

* Static methods require the assembly name in the call. A `static` method and ...
  * ✔️ `invokeMethodAsync('{ASSEMBLY NAME}', '{.NET METHOD ID}', {ARGUMENTS})`
  * ❌ ***NOT*** ... `invokeMethodAsync('{.NET METHOD ID}', {ARGUMENTS})`

* Instance methods don't allow the assembly name in the call. A non-`static` method and ...
  * ✔️ `invokeMethodAsync('{.NET METHOD ID}', {ARGUMENTS})`
  * ❌ ***NOT*** ... `invokeMethodAsync('{ASSEMBLY NAME}', '{.NET METHOD ID}', {ARGUMENTS})`

@contractcooker originally (maybe) said (if I'm not misinterpreting) that some non-`static` instance calls might require the assembly name ***optionally***, but I haven't seen that in testing or from examples sent over by the PU.

The line that I'm adding here to the *Invoke an instance .NET method* section will read (in a NOTE) ...

> \> [!NOTE]
> \> \`invokeMethodAsync\` and \`invokeMethod\` don't accept an assembly name parameter when invoking an instance method.

... so obviously if there are optional cases, then that line shouldn't be added. We'll need to clarify any optional cases. Today, all instance method examples will have ***no*** assembly name parameter passed.

**Comment by guardrex** [comment]
@MackinnonBuck ... I'm going to go ahead now and get this in and live because I have a 🪲 to squash in the example ... and a rotten ***spelling error*** 😈. I'm 👂 for your feedback on that ☝️, and I'll patch things up quickly 🏃 if changes are needed based on your feedback.

File: aspnetcore/blazor/javascript-interoperability/call-dotnet-from-javascript.md [hunk 1]
```diff
@@ -162,7 +162,25 @@ Call `DotNet.createJSStreamReference(streamReference)` to construct a JS stream
...
-* Invoke a .NET instance method from JS using `invokeMethodAsync` or `invokeMethod` (Blazor WebAssembly only) from the passed <xref:Microsoft.JSInterop.DotNetObjectReference>. The .NET instance can also be passed as an argument when invoking other .NET methods from JS.
+* Invoke a .NET instance method from JS using `invokeMethodAsync` (*Recommended*) or `invokeMethod` (Blazor WebAssembly only) from the passed <xref:Microsoft.JSInterop.DotNetObjectReference>. Pass the identifier of the instance .NET method and any arguments. The .NET instance can also be passed as an argument when invoking other .NET methods from JS.
+
+  In the following example:
+
+  * The `{.NET METHOD ID}` placeholder is the .NET method identifier.
+  * The `{ARGUMENTS}` placeholder are optional, comma-separated arguments to pass to the method, each of which must be JSON-serializable.
+
+  ```javascript
+  DotNet.invokeMethodAsync('{.NET METHOD ID}', {ARGUMENTS});
```
Comment (MackinnonBuck) [inline aspnetcore/blazor/javascript-interoperability/call-dotnet-from-javascript.md:173]: Hm... the `DotNet` object can only be used to invoke static methods, not instance methods. Did you intend for this example to be calling `invokeMethodAsync` on a `DotNetObjectReference`?

Comment (MackinnonBuck) [reply aspnetcore/blazor/javascript-interoperability/call-dotnet-from-javascript.md:173]: For instance, the example later in the document using `dotNetHelper` is correct (with your fix, that is 🙂).

**Review by MackinnonBuck — COMMENTED** [review]
Looks good, just a minor correction that I believe was a typo!

**Comment by MackinnonBuck** [comment]
@guardrex, the assembly name is not optional when invoking a static method. Maybe @contractcooker was saying that the assembly name is not always required (i.e., in the case of invoking an instance method), so thus is "optional". Although it really isn't optional, it's either always required (for static methods) or never required (for instance methods).

**Comment by guardrex** [comment]
Good ... that's what I thought.

It looks like I just need to patch that one bit. I'll take care of that right now.

**Pull request merged**